tests/python/pants_test/base/test_hash_utils.py

```diff
@@ -121,8 +120,7 @@ def test_collection_ordering(self):
     self.assertEqual(self._coercing_json_encode({2, 1, 3}), '[1, 2, 3]')
     self.assertEqual(self._coercing_json_encode({'b': 4, 'a': 3}), '{"a": 3, "b": 4}')
     self.assertEqual(self._coercing_json_encode([('b', 4), ('a', 3)]), '[["b", 4], ["a", 3]]')
-    self.assertEqual(self._coercing_json_encode([{'b': 4, 'a': 3}]),
-                     '[{"b": 4, "a": 3}]' if PY3 else '[{"a": 3, "b": 4}]')
+    self.assertEqual(self._coercing_json_encode([{'b': 4, 'a': 3}]), '[{"b": 4, "a": 3}]')
```

Comment:
There must be a fun story behind this one :) Would love to hear it if you have some time. Good change though.

Reply:
Dictionaries in Python 2 are deterministic but only as an implementation detail and not in a particular order. In Python 3.6+, they are deterministic by insertion order. This was the cause of a whole bunch of test failures during the migration :)